### PR TITLE
feat: validate form values

### PR DESCRIPTION
**Pull request by ghost**
1. validate form
2. integrate antd with formik
3. some ui improvements

### CODE REVIEW

File: apps/ckit-app/src/hooks/useFormValidate.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      try {
+        provider.parseToScript(value);
+      } catch (e) {
+        return 'invalid ckb address';
+      }
+    },
+    [provider],
+  );
+
+  const isLockscriptACP = useCallback(
```
Comment (homura) [inline apps/ckit-app/src/hooks/useFormValidate.tsx:29]:
```suggestion
  // TODO refactor with ScriptManager
  const isLockscriptACP = useCallback(
```

File: apps/ckit-app/src/components/Assets/AddAssetButton.tsx [hunk 2]
```diff
@@ -57,124 +58,160 @@ export const ModalForm: React.FC<ModalFormProps> = (props) => {
...
+      const newAssetMeta: AssetMeta = {
+        symbol: values.symbol,
+        decimal: values.decimal,
+        script: {
+          code_hash: values.codeHash,
+          hash_type: values.hashType,
+          args: values.args,
+        },
+      };
+      setAssets(assets.concat(newAssetMeta));
```
Comment (homura) [inline apps/ckit-app/src/components/Assets/AddAssetButton.tsx]:
```suggestion
      addCustomAsset(newAssetMeta);
```

File: apps/ckit-app/src/hooks/useFormValidate.tsx (hunk 1 above)
Comment (ghost) [reply apps/ckit-app/src/hooks/useFormValidate.tsx:29]: fixed

File: apps/ckit-app/src/components/Assets/AddAssetButton.tsx (hunk 2 above)
Comment (ghost) [reply apps/ckit-app/src/components/Assets/AddAssetButton.tsx]: fixed

**Pull request merged**